Makefile

```diff
@@ -98,14 +101,11 @@ dev-push:
 ###################################
 
 # Create YAML file for deployment
-prod-yaml:
-ifeq ($(KUSTOMIZE_VERSION),$(filter $(KUSTOMIZE_VERSION),unknown v1))
-	@echo "please upgrade kustomize version to v2 at least."
-else
+prod-yaml: | $(KUSTOMIZE)
+	@$(KUSTOMIZE) version 2>&1 | grep -q 'KustomizeVersion:v\{0,1\}[2-9][[:digit:]]\{0,\}' || { echo "kustomize v2+ required" 1>&2; exit 1; }
```

Comment:
If we're dictating the version of kustmoize, do we still need these version checks?

Reply:
Hi @andrewsykim,

Sí, because the only time Kustomize is installed to a sub-directory of the project is if `command -v kustomize` returns a non-zero exit code. Meaning we'll use an existing version of Kustomize if it's present.

Reply:
On linux, `kustomize version` doesn't have the `v` prefix which breaks the Makefile:
```
$ kustomize version
Version: {KustomizeVersion:2.0.3 GitCommit:a6f65144121d1955266b0cd836ce954c04122dc8 BuildDate:2019-03-05T20:37:42Z GoOs:linux GoArch:amd64}
```

Does this account for that?

Reply:
```

Reply:
Sí, that's what `v\{0,1\}` does. It says either zero or one `v`.

**san v**
```shell
$ cat <<EOF | grep -o 'KustomizeVersion:v\{0,1\}[2-9][[:digit:]]\{0,\}'
> $ kustomize version
> Version: {KustomizeVersion:2.0.3 GitCommit:a6f65144121d1955266b0cd836ce954c04122dc8 BuildDate:2019-03-05T20:37:42Z GoOs:linux GoArch:amd64}
> EOF
KustomizeVersion:2
```

**cum v**
```shell
$ cat <<EOF | grep -o 'KustomizeVersion:v\{0,1\}[2-9][[:digit:]]\{0,\}'
> $ kustomize version
> Version: {KustomizeVersion:v2.0.3 GitCommit:a6f65144121d1955266b0cd836ce954c04122dc8 BuildDate:2019-03-05T20:37:42Z GoOs:linux GoArch:amd64}
> EOF
KustomizeVersion:v2
```

Reply:
thanks

Reply:
FWIW, the regex says `KustomizeVersion:` followed by an optional `v` followed by a digit that is `2-9` followed one or more additional, optional digits.

Basically, it has to be at least version 2 *something*, but we don't care what.

Reply:
Hope this doesn't break with kustomize v10.* :trollface: